test/controllers/api/nodes_controller_test.rb

```diff
@@ -138,7 +138,7 @@ def test_create_invalid_xml
       xml = "<osm><node lat='#{lat}' lon='#{lon}' changeset='#{changeset.id}'><tag k='foo' v='#{'x' * 256}'/></node></osm>"
       put node_create_path, :params => xml, :headers => auth_header
       assert_response :bad_request, "node upload did not return bad_request status"
-      assert_equal ["NodeTag ", " v: is too long (maximum is 255 characters) (\"#{'x' * 256}\")"], @response.body.split(/[0-9]+,foo:/)
```

Comment:
It's CPK related - the new rails support manages to include the node ID and version in the message while with the old CPK module it failed to add the ID and you just got a space in the message. As the ID can vary it was easy just to match the message which is what we care about.

Reply:
But if the output from the API changes, will that break client apps that are parsing the previous output?

Reply:
I remembered the details slightly wrong... This is what it used to look like:

```
NodeTag 14,foo: v: is too long (maximum is 255 characters)
```

and now:

```
NodeTag [38, "foo"]: v: is too long (maximum is 255 characters)
```

which is all down to how rails default error handling (augmented by CPK in the old code) represents a composite key in a validation failure which I'm not sure we can do much about?

Are users really supposed to be parsing this?

Reply:
FWIW: I invented my own error messages in cgimap for keys/values exceeding 255 chars, and nobody seemed to have an issue with it. That's for the diff upload, though.

Error messages are: _Key has more than 255 unicode characters_ and _Value has more than 255 unicode characters_.

Reply:
I know that some clients do parse the free-text error responses (see e.g. https://github.com/openstreetmap/openstreetmap-website/issues/2227 inter alia). 

But perhaps for this particular error it's very unlikely that clients would be running into it (other than during development) since it's very likely that adding the length limit would happen client-side. It's a different situation for other messages, like when an editing conflict happens (e.g. version mismatch) which you can't handle 100% client side.

So I think it's worth us being aware that this is technically a breaking change, but we don't expect (m?)any people to notice, and if there is a problem we'll need to build some method to fully control the errors in future (i.e. not rely on rails generated error messages since these can change between versions).